subsys/net/lib/sockets/sockets.c

```diff
@@ -143,11 +143,12 @@ static void zsock_received_cb(struct net_context *ctx, struct net_pkt *pkt,
 	/* Normal packet */
 	net_pkt_set_eof(pkt, false);
 
-	/* We don't care about packet header, so get rid of it asap */
-	header_len = net_pkt_appdata(pkt) - pkt->frags->data;
-	net_buf_pull(pkt->frags, header_len);
-
 	if (net_context_get_type(ctx) == SOCK_STREAM) {
+		/* TCP: we don't care about packet header, get rid of it asap.
```

Comment:
Why this conditionalizing, leading to code duplication? If we (may) need a packet header, let's keep it here, and consistently remove in zsock_recv[from]().

Reply:
Initially, I prefer to keep the packet header for both cases. But I found zsock_recv_stream() needs to be modified a lot because packet header isn't dropped. I will modify this to prefer consistency.

Reply:
Hmm, you may be right - and we can't return src addr for an arbitrary TCP recv anyway (only for recv() which starts to consume a new packet). Well, that's why it's useful to know why something was implemented this or that way, so having extra info in the commit messages or PR comments is helpful. And yeah, I'd appreciate your giving another thought to trying to avoid that code duplication - if it will lead to only more code being added, sure, it's ok to leave it like that.